delta-api/src/main/java/io/cdap/delta/api/DDLOperation.java

```diff
@@ -16,15 +16,59 @@
 
 package io.cdap.delta.api;
 
+import java.util.Objects;
+import javax.annotation.Nullable;
+
 /**
- * A DDL operation
+ * Represents DDL operation.
  */
-public enum DDLOperation {
-  CREATE_DATABASE,
-  DROP_DATABASE,
-  CREATE_TABLE,
-  DROP_TABLE,
-  TRUNCATE_TABLE,
-  ALTER_TABLE,
-  RENAME_TABLE
+public class DDLOperation {
+  /**
+   * Type of DDL Operation
+   */
+  public enum Type {
+    CREATE_DATABASE,
+    DROP_DATABASE,
+    CREATE_TABLE,
+    DROP_TABLE,
+    TRUNCATE_TABLE,
+    ALTER_TABLE,
+    RENAME_TABLE
+  }
+
+  private final String tableName;
```

Comment:
I think it makes sense for this to also have the prevTableName field, as a RENAME_TABLE operation will need it. I would assume we need to update the ddl count for both the previous table name and the new one?

Reply:
added prevTableName for rename operation.